### PR TITLE
Upgrade @angular-eslint/*

### PR DESCRIPTION
**Pull request by vitaliibendyk**
BREAKING CHANGE:
Requires angular 19.

### CODE REVIEW

File: packages/eslint-config-recommended/package.json [hunk 1]
```diff
@@ -37,19 +37,19 @@
...
     "eslint-plugin-import": "^2.28.1",
     "eslint-plugin-unused-imports": "^3.2.0"
   },
   "peerDependencies": {
-    "@angular-eslint/builder": "^18.0.0",
-    "@angular-eslint/eslint-plugin": "^18.0.0",
-    "@angular-eslint/eslint-plugin-template": "^18.0.0",
-    "@angular-eslint/schematics": "^18.0.0",
-    "@angular-eslint/template-parser": "^18.0.0"
+    "@angular-eslint/builder": "^19.0.2",
```
Comment (michaelwittwer) [inline packages/eslint-config-recommended/package.json:48]: if there is no explicit reason on why `19.0.0` & `19.0.1` are not allowed, I would just use `^19.0.0` as peerDep definition.

File: packages/eslint-config-recommended/package.json [hunk 2]
```diff
@@ -37,19 +37,19 @@
   },
   "dependencies": {
     "@shiftcode/eslint-plugin-rules": "^2.0.1",
-    "@typescript-eslint/eslint-plugin": "^7.16.0",
-    "@typescript-eslint/parser": "^7.16.0",
+    "@typescript-eslint/eslint-plugin": "^8.16.0",
```
Comment (michaelwittwer) [inline packages/eslint-config-recommended/package.json:40]: use the latest versions here and next line

**Review by michaelwittwer — CHANGES_REQUESTED** [review]
The commit message is a bit misleading, since this change has not directly todo with angular@19 but instead with @angular-eslint/* packages, which then use angular@19 under the hood, but let's keep it like that
@michaelwittwer I based my commit message on your analog
[
Screenshot 2024-12-11 at 11 37 05
](url)

**Comment by michaelwittwer** [comment]
@vitaliibendyk did you check the [CHANGELOG](https://github.com/typescript-eslint/typescript-eslint/blob/main/CHANGELOG.md#800-2024-07-31) and [Announcing Typescript EsLint 8](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8/) to make sure there are no additional changes necessary?

**Comment by michaelwittwer** [comment]
@vitaliibendyk every new version of `@angular-eslint` brings [changes to which rules are enabled](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8/#updated-configuration-rules), make sure to test this version in an existing project to know if we need to update our rules configuration.

**Comment by michaelwittwer** [comment]
@vitaliibendyk please also update the `@angular-esling/*` packages in our `eslint-plugin-rules` package.

File: packages/eslint-config-recommended/package.json (hunk 1 above)
Comment (vitaliibendyk) [reply packages/eslint-config-recommended/package.json:48]: @michaelwittwer I based my commit message on your analog 
Screenshot 2024-12-11 at 11 37 05

**Comment by michaelwittwer** [comment]
> The commit message is a bit misleading, since this change has not directly todo with angular@19 but instead with @angular-eslint/* packages, which then use angular@19 under the hood, but let's keep it like that @michaelwittwer I based my commit message on your analog [ Screenshot 2024-12-11 at 11 37 05 ](url)

Thanks for pointing it out. In the end it's true, but not the full story. 🙈 As i said, leave it like that.

**Pull request closed without merging**